src/service.js

```diff
@@ -292,49 +255,34 @@ export function getServicePromiseOrNull(win, id) {
   return getServicePromiseOrNullInternal(win, id);
 }
 
-
 /**
  * Returns a service for the given id and ampdoc (a per-ampdoc singleton).
  * If the service is not yet available the factory function is invoked and
  * expected to return the service.
  * @param {!Node|!./service/ampdoc-impl.AmpDoc} nodeOrDoc
  * @param {string} id of the service.
  * @param {function(!./service/ampdoc-impl.AmpDoc):T=} opt_factory
- *     Should create the service if it does not exist yet. If the factory is
- *     not given, it is an error if the service does not exist yet.
+ *     Factory to create the service if the service does not exist yet.
```

Comment:
There are `getExistingService*orNull` functions, but I've completely removed functions that will fail an assertion if a service doesn't already exist as `getServiceInternal` now handles that case.

There may not be some `forDoc` ones as they weren't needed anywhere. I can add some in if you'd like. 

As for `performaneForOrNull`, my understanding is that there are times we don't expect the performance service to exist. If that's not the case then we can remove `performanceForOrNull` as @choumx suggested.